### DISCUSSION TITLE
Exposing Metadata Issue

### DISCUSSION THREAD

**Discussion by lavarocker** [savonet/liquidsoap#3320] [User Support]
Hello,

I have a simple script that runs two radios playing different selections and relays through Icecast running on the same VM. 

I want to push the metadata to the localhost webserver and scrape it onto a website.  I added the following code from the Liquidsoap book to my script for the first radio:

          _last_metadata = ref([])
          radio.on_track(fun (m) -> last_metadata := m)
          def show_metadata(~protocol, ~headers, ~data, uri)
          http.response(content_type="application/json; charset=UTF-8",
          data=metadata.json.stringify(!last_metadata))
          end
          harbor.http.register("/metadata", show_metadata_

My radio is named 'radio' so no changes were made.  I get the following errors when running the script:

          _At ./new-replaygain.liq, line 32 char 0 - line 35 char 3:
          def show_metadata(~protocol, ~headers, ~data, uri)
          http.response(content_type="application/json; charset=UTF-8",
          data=metadata.json.stringify(!last_metadata))
          end
           Warning 4: Unused variable data_

Anyone see what I'm missing here?  If I can get this to work I want to push metadata from both radios to the webserver.

Any help is much appreciated.

Rocker

Full script is attached (passwds/names hidden)
---------------------------------------------------------------------------------------------------
[liquidsoap test script.txt](https://github.com/savonet/liquidsoap/files/12352201/liquidsoap.test.script.txt)

**Comment by toots** [comment]
Hi @lavarocker This is because the variable that comes from the function parameters is not used in your script. Two things you can:

* For labelled arguments, you can mark them as ignored:
```liquidsoap
      def show_metadata(~protocol, ~headers, ~data=_, uri)
        http.response(
          content_type="application/json; charset=UTF-8",
          data=metadata.json.stringify(!last_metadata)
        )
      end
```

* In general, you can use the `ignore` operator:
```liquisoap
      def show_metadata(~protocol, ~headers, ~data, uri)
        ignore(data)
        http.response(
          content_type="application/json; charset=UTF-8",
          data=metadata.json.stringify(!last_metadata)
        )
      end
```

**Reply by lavarocker** [reply]
I will try that - thank you very much.  I'm really enjoying my Liquidsoap radios and looking forward to adding new features to the scripts.

A question about the integrated web server:  how robust is it?  Can it handle direct user metadata requests or should I have the webserver make those requests and pass them to the users?  I don't have a lot of listeners and will keep it that way.
